### DISCUSSION TITLE
Can i do tests via Foundry tools if protocol using Hardhat?

### DISCUSSION THREAD

**Discussion by moray5554** [Cyfrin/security-and-auditing-full-course-s23#55] [Q&A] [answered]
Can i do tests via Foundry tools if protocol using Hardhat? I mentioned some projects building with Hardhat and don`t know how have i test it. Should i use some hardhat-foundry plugins for them? Or how you guys approaching those things?

**Comment by moray5554** [comment]
upd. peeps told me - best practice is to know both of them and use it depends on which one protocol was build

**Comment by PatrickAlphaC** [comment, marked as answer]
Hi @moray5554!

Great question,

And the answer is that it depends on what you want to do! 

For me, I sometimes re-write tests in foundry to play with a codebase. But sometimes, just knowing hardhat makes it easier. It's really up to you!
